### PR TITLE
Updated regal to latest version

### LINKED ISSUE #917: Update Regal to latest version
# 💡 Summary #

Update run_opa_tests workflow to use latest regal actions. 

## Motivation and context ##

Why does this work belong in this project?

This would be useful because Regal is currently on version 0.16.0 and we are using 0.11.0.  

Running the workflow yields the following annotation: 
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: StyraInc/setup-regal@v0.2.0. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.

Node 16 has reached its [end of life](https://github.com/nodejs/Release/#end-of-life-releases),

May not resolve issue, may need support from Styra Inc.  See [issue](https://github.com/StyraInc/regal/issues/559).

## Implementation notes ##

Please provide details for implementation, such as:

- an example for how this would be used
- what this would look like
- how this would act
- any related work, including links to related issues

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Criterion

### PR DESCRIPTION
**Pull request by crutchfield**
## 🗣 Description ##

Updated regal to use latest version - update uses Node 20+ 
closes #917 

## 💭 Motivation and context ##

Node 16 is no longer supported with security updates

## 🧪 Testing ##

- Executed Rego unit test / linter 

## ✅ Pre-approval checklist ##

- [X] This PR has an informative and human-readable title.
- [X] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [X] Changes are limited to a single goal - *eschew scope creep!*
- [X] Changes are sized such that they do not touch excessive number of files.
- [X] *All* future TODOs are captured in issues, which are referenced in code comments.
- [X] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [X] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [X] All relevant type-of-change labels added.
- [X] All relevant project fields are set.
- [X] All relevant repo and/or project documentation updated to reflect these changes.
- [X] Unit tests added/updated to cover PowerShell and Rego changes.
- [X] Functional tests added/updated to cover PowerShell and Rego changes.
- [X] All relevant functional tests passed.
- [X] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Review by james-garriss — APPROVED** [review]
Looks like minor changes for a language version update.

File: PowerShell/ScubaGear/Rego/DefenderConfig.rego [hunk 1]
```diff
@@ -23,21 +23,24 @@ import data.utils.defender.ApplyLicenseWarning
...
+ReportDetails1_1(false, false) := "Standard and Strict preset policies are both disabled"
 
 # Parse through all items in Policies, if item identity is the one
 # we want & state is enabled, save item. Return number of items saved.
-GetEnabledPolicies(Policies, Identity) := count([Policy |
-    some Policy in Policies
-    Policy.Identity == Identity
-    Policy.State == "Enabled"
-]) > 0
+#default GetEnabledPolicies := false
```
[the remark this answers is not in the document]
Comment (crutchfield) [reply PowerShell/ScubaGear/Rego/DefenderConfig.rego:36]: Removed

**Comment by crutchfield** [comment]
> Reviewed changed and ran `Invoke-Scuba` which worked as intended with latest release. However, noted that changing import rego.v1 and removing `if {}` statements has the effect of making versions of Rego <0.59 fail as the syntax is not supported by earlier versions of the engine. I am okay with requiring newer versions, but it means we should also update the list of allowed OPA rego engine versions in the Install-OPA method to only support 0.59 and up (0.60 and 0.61 at the moment).
> 
> Recommend updating allowed list by removing those now unsupported versions. Also recommend rebase first as the allow list has been updated by a now merged commit.

Change to allow tested version 59-61.

**Comment by crutchfield** [comment]
@nanda-katikaneni  PR is ready to merge to main

**Pull request merged**